types/params.go

```diff
@@ -193,6 +197,10 @@ func (params ConsensusParams) ValidateBasic() error {
 		return errors.New("len(Validator.PubKeyTypes) must be greater than 0")
 	}
 
+	if len(params.Validator.PubKeyTypes) > 1 {
```

Comment:
I would merge the 2 conditions together:

```suggestion
	if len(params.Validator.PubKeyTypes) != 1 {
```